src/backend/processor/diagnostics.ts

```diff
@@ -76,6 +76,13 @@ export class DiagnosticsApi {
             filesToLoad.push(this._selectedEntry.file);
         }
 
+        // Unload reports without calling parse
+        if (filesToLoad.length === 0) {
+            const dummyData: ParseResult = { reports: [], version: 1 };
+            this.parseDiagnosticsData(JSON.stringify(dummyData));
```

Comment:
I don't really like this approach that we are creating a dummy data only to call this function. Why do we need to call `parseDiagnosticsData` here? To initialize `this._diagnosticEntries` and other variables? In this case we can initialize these fields here too. Also if it contains some extra logic we can create a separate function for it and we can call it here and in the `parseDiagnosticsData` function.

Reply:
Moved the field initializations here.

The parseMetadata does a little bit of extra processing when deciding whether this._selectedEntry should be kept, but selectedEntry cannot exist inside the unloading if, so it's no concern here.